Write conclusion (#101)
daffidwilde/thesis: chapters/conclusion.tex

datasets exist for an algorithm according to some metric. This exploration is
achieved via a bespoke evolutionary algorithm which acts on datasets of unfixed
shapes, sizes and data types. The chapter presented descriptions and
illustrations of the internal mechanisms of the EDO method, as well as briefly
describing a Python implementation. Finally, the chapter concluded with an
extensive case study, demonstrating the capabilities and nuances of EDO in
gaining a richer picture of an algorithm's abilities independently, and against
a competitor.

Following the discussion of `fair' machine learning practices in the literature
review, Chapter~\ref{chp:kmodes} offered a novel initialisation to the
\(k\)-modes algorithm which made use of game theory. The new initialisation
extended a commonly used method, but replaced its greedy component with a
solvable matching game. In the evaluative section of this chapter, traditional
assessment techniques suggested that the new method improved upon the original,
and so the original was discarded.

However, the new method did not consistently outperform another well-known
initialisation. To better understand the conditions under which either of the

> remaining initialisations would succeed, a similar competitive setting to

Reviewer: ```suggestion
remaining initialisations would succeed, a similar setting to
```

clustering is included. In general, pursuits of this kind reformulate existing
techniques to be mathematically fair. The initialisation presented in
Chapter~\ref{chp:kmodes}, instead, incorporated game theory directly into an
existing algorithm. In doing so, an improvement over the existing method was
shown, using both traditional confirmation processes and the EDO method.

The framework used in Chapter~\ref{chp:copd} contributed to healthcare modelling
literature in three ways. First, the estimation of queuing parameters via the
Wasserstein distance has expanded a relatively scarce aspect of queuing
research. Second, by making COPD the subject of the methodology, the framework
has added to a body of literature surrounding a condition that is vital to
understand given its prevalence, as well as its links to deprivation and
comorbidity. Lastly, the framework provided a solution to the common issue of
data availability in modern operational research. By combining the various
individual methods, valuable insights were extracted from a relatively
unsophisticated data source, which is a result seldom seen in operational

> research.

Reviewer: Nice. :+1:

In addition to the work directly included in the chapters of this thesis, the
research associated with this thesis has resulted in the production of numerous
auxiliary research items. These include several well-developed pieces of
research software, and a number of useful, publicly available datasets for

> clustering and healthcare modelling.

Reviewer: Can we point at them (listed in brackets is fine, or perhaps this needs to be a standalone page with everything listed, I forget if you have that already?).

Author: I don't have one for the datasets, but I do list the DOI names for them in the introduction. The software packages are summarised in the dissemination.

Do you think another page would be helpful?

Reviewer: It's probably fine to have those two spots in which case can you just point to them here? Something like:

```
research software (listed in \ref{dissemination}, and a number of useful, publicly available datasets (listed in Section \ref{}).
```

Author: Cool cool. Added in those references in ac10058

\section{Further work}

\subsection*{EDO as a data synthesiser}

As demonstrated in the case study in Chapter~\ref{chp:edo} and the closing
section of Chapter~\ref{chp:kmodes}, the EDO method is capable of facilitating
richer insights into an algorithm's performance. Having said that, a limitation
of the method is that there is no standardised way to guarantee relationships

> between different columns in a dataset, or the families passed to EDO.

Reviewer: ```suggestion
between different columns in a dataset (\(\mathcal P\), or the families passed to EDO.
```

parameters of the distribution families would be relatively straightforward, but
incorporating the relationships between them is less so.

This capability has been one of the major attractions of using GANs for data
synthesis, but their black-box nature defeats the object of EDO. Another option
is to use copulas. Copulas are functions that join multivariate distribution
functions to their one-dimensional margins~\cite{Nelsen1999}. For EDO, this
would mean \(\mathcal P\) would contain a single element: a copula function
fitted to the existing dataset. In this case, the technical aspects of an
individual's representation would need adjusting to accommodate this change.
Likewise, the crossover and mutation processes would require some changes to
account for the lack of distinct distribution families.

A Python implementation of copulas for data synthesis exists~\cite{copulas}, and
incorporating this as a dependency of the \edo\ library would reduce the work
required to implement this feature. Studying the impact of copulas in EDO would
provide a valuable opportunity to demonstrate the capabilities of EDO as a fully

> fledged data synthesis method.

Reviewer: That's neat!

Reviewer: You should open this up as an issue on https://github.com/daffidwilde/edo/issues

Author: Will do! 😄

Author: Done.

deeply varied and heterogeneous --- as expected. However, the dataset was
insufficiently detailed to construct meaningful models on the entire population
or system. The variety in the data opened up an interest in population
segmentation techniques, eventually feeding into
Chapters~\ref{chp:kmodes}~and~\ref{chp:copd}. Meanwhile, requests were made for
larger, more detailed datasets --- so that contemporary, machine learning
techniques could be applied more readily --- and the gathering of literature
began. Even with the expansive nature of machine learning literature, two clear
patterns emerged.

First, the vast majority of publications that introduced a novel machine
learning method contained a boilerplate evaluation section. In each article, the
proposed method would be pitted against a few of its contemporaries by comparing
a handful of metrics on a handful of datasets. Typically, these metrics and
datasets would be taken from a small pool which was relevant to that technique;
this makes sense. There are appropriate measures for various techniques, and it
is important to make comparisons relative to a fixed point.

> However, this approach to algorithm evaluation exhibited two issues: one, a lack

Reviewer: ```suggestion
This approach to algorithm evaluation exhibited two issues: one, a lack
```

of diversity in the resources used to assess algorithm performance; and, two, an
incongruence between the power of the evaluation process and the conclusions
drawn from that process. Often, a method would be deemed `state-of-the-art' or
`better' based exclusively on a process that offered little insight into the
actual quality of that method. This reliance on a narrow assessment process
prompted research into how else an algorithm could be evaluated objectively,
directly leading to the work in Chapter~\ref{chp:edo}.

The second observation was that the ethics of data and machine learning
algorithms were being discussed, but the discourse appeared separately from the

> machine learning publications. An exception to this was the development fair

Reviewer: ```suggestion
machine learning publications. An exception to this was the development of fair
```

machine learning practices. These methods consider new formulations and
objectives based on mathematical fairness, a concept either derived from or with
common roots in game theory. The reinventing of techniques and paradigms to be
fair raised questions about how far an addition of some game theory could
improve the performance of an existing algorithm, as opposed to creating an
entirely new one, leading to the work in Chapter~\ref{chp:kmodes}.

Over the course of this project, it became clear that the requests for more
detailed datasets would not complete in time. Hence, the research methodology
would have to prioritise extracting as much useful information from the data at
hand. Simultaneously, \ctmuhb\ provided further administrative datasets which
related to the members of their population who suffer from the respiratory
condition, COPD. An internal report by NHS Wales found that \ctmuhb\ had the
highest prevalence of the condition out of all the Welsh health boards. Given
the known links between COPD, socioeconomic deprivation and the coincidence of
multiple comorbidities, the objective of this project was revised to focus on
understanding the needs of that population.

> Each of these concerns culminated in the methodology presented in

Reviewer: I wonder if we can use a softer word than "concerns", perhaps "challenges"?

```suggestion
Overcoming each of these challenges culminated in the methodology presented in
```

Author: A good point well made. I'll make these changes now 👍
